kong/plugins/ip-restriction/handler.lua

```diff
@@ -29,6 +33,31 @@ do
 end
 
 
+local is_http_subsystem = ngx.config.subsystem == "http"
+
+
+local do_exit
+if is_http_subsystem then
+  do_exit = function(status, message)
+    return kong.response.error(status, message)
+  end
+
+else
+  do_exit = function(status, message)
+    local tcpsock, err = ngx_req.socket(true)
+    if err then
+      error(err)
+    end
+
+    tcpsock:send(cjson_encode({
+      status  = status,
+      message = message
+    }))
+
+    return ngx_exit(1)
```

Comment:
sorry, I misled you, maybe `ngx_exit(status)` would be better here, right? cc @dndx